java-frontend/src/main/java/org/sonar/java/resolve/JavaType.java

```diff
@@ -492,8 +512,14 @@ public boolean isSubtypeOf(Type superType) {
     public boolean isSubtypeOfBound(JavaType type) {
       switch (boundType) {
         case SUPER:
+          if(bound.isTagged(TYPEVAR) && !type.isTagged(TYPEVAR) && bound != type) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)